src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1625,7 +1625,13 @@ private AuthenticationResult refreshToken(final CallbackHandler callbackHandle,
                 // remove item from cache to avoid same usage of
                 // refresh token in next acquireToken call
                 removeItemFromCache(refreshItem);
-                return acquireTokenLocalCall(callbackHandle, activity, useDialog, request);
+                if (!request.isSilent() && callbackHandle.callback != null && activity != null) {
+                    return acquireTokenLocalCall(callbackHandle, activity, useDialog, request);
+                } else {
+                    callbackHandle.onError(
+                            new AuthenticationException(ADALError.AUTH_FAILED_SERVER_ERROR, request.getLogInfo() + errLogInfo));
```

Comment:
Can we return the AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED error code? This is the error code we advertise to all the adal users for silent request fails, and you need a interactive flow to start again. Changing the error code will break the rule we already set.